src/standalone/Selector/MultiSelectWithTags.tsx

```diff
@@ -0,0 +1,302 @@
+import React, { useState, useCallback, useRef } from "react";
+import { SelectorData, SelectorPropsSingleSelect } from "./Selector";
+import SingleSelect from "./Selector";
+import { TextField, TextFieldProps, Tooltip } from "@material-ui/core";
+import { createStyles, Theme, withStyles, WithStyles } from "@material-ui/core";
+import { SmallIconButton, SmallListItemIcon } from "../Small/List";
+import { Search as SearchIcon, Info as InfoIcon } from "@material-ui/icons";
+import { RemoveIcon } from "../../standalone";
+import { Autocomplete } from "@material-ui/lab";
+import { GenericWithStyles } from "../../utils";
+
+export interface MultiSelectData extends SelectorData {
+	/**
+	 * Item type value
+	 */
+	type?: string;
+	/**
+	 * Item click handler
+	 */
+	onClick?: () => void;
+	/**
+	 * Can the entry be unselected?
+	 * @param data The data entry to be unselected
+	 */
+	canUnselect?: (data: MultiSelectData) => boolean | Promise<boolean>;
+}
+
+export interface MultiSelectWithTagsProps<Data extends MultiSelectData>
+	extends Omit<
+		SelectorPropsSingleSelect<Data>,
+		"multiSelect" | "clearable" | "onSelect" | "selected"
+	> {
+	/**
+	 * Extended selection change handler
+	 * @param data The selected data entries
+	 */
+	onSelect?: (value: Data[]) => void;
+	/**
+	 * Extended group selection change handler
+	 * @param data The selected data entry
+	 */
+	onGroupSelect?: (value: SelectorData | null) => void;
+	/**
+	 * The title of control
+	 */
+	title: string;
+	/**
+	 * The currently selected values
+	 */
+	selected: Data[];
+	/**
+	 * The filtered data after selected dropdown value
+	 */
+	filteredData: Data[];
+	/**
+	 * Set filtered data after selected from search input
+	 */
+	setData?: (values: Data[]) => void;
+	/**
+	 * The currently selected groups
+	 */
+	selectedGroup: SelectorData | null;
+	/**
+	 * Callback method on group selection load
+	 */
+	onGroupLoad: (search: string) => Data[] | Promise<Data[]>;
+	/**
+	 * Label for search input control
+	 */
+	searchInputLabel?: string;
+}
+
+const styles = createStyles((theme: Theme) => ({
+	paperWrapper: {
+		boxShadow: "none",
+		border: `1px solid ${theme.palette.divider}`,
+	},
+	outlined: {
+		float: "left",
+		backgroundColor: "#cce1f6",
+		padding: "0px 20px",
+		borderRadius: 20,
+		borderColor: "#cce1f6",
+		margin: "5px",
+		lineHeight: "30px",
+	},
+}));
+
+const MultiSelectWithTags = <Data extends MultiSelectData>(
+	props: MultiSelectWithTagsProps<Data> & WithStyles
+) => {
+	const {
+		title,
+		searchInputLabel,
+		onSelect,
+		selected,
+		filteredData,
+		setData,
+		onGroupLoad,
+		onGroupSelect,
+		selectedGroup,
+		enableIcons,
+		disable,
+		classes,
+	} = props;
+	const [selectedValues, setSelectedValue] = useState<Data[]>([]);
+	const input = useRef<TextFieldProps>();
+	const array = [...selected, ...selectedValues].flat();
+	const allSelected: Data[] = [];
```

Comment:
you may need to disable the eslint rule which prevents usage of the non-null assert (`!`)
```ts
const allSelected = uniqueArray(array.map(item => item.value))
                                  .map(value => array.find(e => e.value === value)!)
```
uniqueArray is a utility function from SCF, you can copy paste it into the src/utils folder (don't forget to add documentation).

Reply:
Util files added as arrayFunctions.